lib/resource.js

```diff
@@ -50,12 +67,34 @@ class Resource {
 		this.filename = filename;
 	}
 
-	getText () {
-		return this.text;
+	async getText () {
+		switch (this.tempMode) {
+			case 'memory':
+				return await this.text;
+			case 'memory-compressed':
+				return (await decompress(this.text)).toString(this.getEncoding());
```

Comment:
Do we need `.toString(this.getEncoding())` here? it looks like we will call `toString` of `string`?